python-checks/src/test/resources/checks/booleanCheckNotInverted.py

```diff
@@ -68,3 +68,23 @@ def func2():
 #                 ^^^^^^^^^^
         b = 10
     return "item1" "item2"
+
+def set_comparison_test():
```

Comment:
I think it would be nice to have some more tests highlighting the assumptions we make: 
* Comparing a known set with ANY type won't raise an issue
* Comparing two ANY types will raise an issue (I know the one related to SONARPY-1574 does that, but only by accident)
* Subtypes of `set` won't raise an issue as well.
* We can also document that we may still have FPs in cases such as (and maybe see if we can fix them without losing too many TPs):
```
def foo(a, cond):
  if cond:
    b = {'nut', 'apple', 'potatos'}
  else:
    b = "hello"
  if not (a <= b): ...
```
* And there may be some other cases like this that could be documented.

Reply:
I added a few test cases in 4f00f494229181d05dd8bb4a811b7777636c400d.